src/main/java/org/mozilla/gecko/background/healthreport/HealthReportDatabaseStorage.java

```
@@ -1031,7 +1039,13 @@ private void recordDailyLast(int env, int day, int field, Object value, String t
       v.put("env", env);
       v.put("field", field);
       v.put("date", day);
-      db.insert(table, null, v);
+      try {
+        db.insertOrThrow(table, null, v);
+      } catch (SQLiteConstraintException e) {
+        throw new IllegalStateException("Event did not reference existing an environment or field.", e);
+      } catch (Exception e) {
+        Logger.error(LOG_TAG, "Unknown exception thrown while recording daily last value.", e);
```

I'd rather have the original exception propagate rather than be logged; just drop this catch clause.